src/hb-ot-cmap-table.hh

```diff
@@ -918,8 +957,14 @@ struct cmap
     // Write out format 12 sub table.
     if (cmap_subset_plan.format12_groups)
     {
-      EncodingRecord &format12_rec = table->encodingRecord[2];
+      if (unlikely (!cmap_subset_plan.has_unicode_ucs4 && !cmap_subset_plan.has_ms_ucs4)) return false;
```

Comment:
Not having a cmap 12 shouldn't be an error as long as you have at least one cmap sub-table.

Reply:
True, but note the conditional `if (cmap_subset_plan.format12_groups)` two lines above. If `format12_groups` exists but no ucs4 cmap tables exist, something must be wrong. I speculate it could still happen say if non-Unicode cmap has format 12 contents unexpectedly. Thus this `if (unlikely` conditional.

Reply:
Ah ok, that makes sense.